src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1433,6 +1435,42 @@ private BoundExpression BindSizeOf(SizeOfExpressionSyntax node, BindingDiagnosti
                 this.GetSpecialType(SpecialType.System_Int32, diagnostics, node), hasErrors);
         }
 
+        private BoundExpression BindFieldExpression(FieldExpressionSyntax node, BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(ContainingType is { });
+            SynthesizedBackingFieldSymbolBase? field = null;
+
+            ReportFieldContextualKeywordConflict(node, node.Token, diagnostics);
+
+            switch (ContainingMember())
+            {
+                case SynthesizedBackingFieldSymbolBase backingField:
+                    field = backingField;
+                    break;
+                case MethodSymbol { AssociatedSymbol: SourcePropertySymbol property }:
+                    field = property.BackingField;
+                    break;
+                default:
+                    {
+                        Debug.Assert((this.Flags & BinderFlags.InContextualAttributeBinder) != 0);
+                        var contextualAttributeBinder = TryGetContextualAttributeBinder(this);
+                        if (contextualAttributeBinder is { AttributeTarget: MethodSymbol { AssociatedSymbol: SourcePropertySymbol property } })
+                        {
+                            field = property.BackingField;
+                        }
+                        break;
+                    }
+            }
+
+            if (field is null)
+            {
+                throw ExceptionUtilities.UnexpectedValue(ContainingMember());
+            }
```

Comment:
```suggestion
            SynthesizedBackingFieldSymbolBase field;

            ReportFieldContextualKeywordConflict(node, node.Token, diagnostics);

            switch (ContainingMember())
            {
                case SynthesizedBackingFieldSymbolBase backingField:
                    field = backingField;
                    break;
                case MethodSymbol { AssociatedSymbol: SourcePropertySymbol property }:
                    field = property.BackingField;
                    break;
                default:
                    {
                        Debug.Assert((this.Flags & BinderFlags.InContextualAttributeBinder) != 0);
                        var contextualAttributeBinder = TryGetContextualAttributeBinder(this);
                        if (contextualAttributeBinder is { AttributeTarget: MethodSymbol { AssociatedSymbol: SourcePropertySymbol property } })
                        {
                            field = property.BackingField;
                        }
                        else
                        {
                            throw ExceptionUtilities.UnexpectedValue(ContainingMember());
                        }
                        break;
                    }
            }
```

Reply:
Yes, that's what I understood your original suggestion to be. But `PropertySymbol.BackingField` is not annotated as returning a non-`null` value so I'd still want to check `field is { }` after the `switch`. (I could check `{ AssociatedSymbol: SourcePropertySymbol { BackingField: { } field }` in each of the two `switch` cases, but that seems redundant.)

Reply:
`SourcePropertySymbolBase.BackingField` is declared in a nullable-disabled context, and if it was annotated, it would be nullable since the value will be `null` for properties other than auto properties.

Reply:
Ah, ok, I was misunderstanding your comment: I was reading as "definitely doesn't return null", and you meant "is null-oblivious". Thanks.